app/boards/shields/bfo9000/bfo9000.dtsi

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020 Pete Johanson, Kurtis Lew
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <dt-bindings/zmk/matrix-transform.h>
+
+/ {
+	chosen {
+		zmk,kscan = &kscan0;
+		zmk,matrix_transform = &default_transform;
+	};
+
+	default_transform: keymap_transform_0 {
+		compatible = "zmk,matrix-transform";
+		columns = <18>;
+		rows = <6>;
+		map = <
+		RC(0,0) RC(0,1) RC(0,2) RC(0,3) RC(0,4) RC(0,5) RC(0,6) RC(0,7) RC(0,8) RC(0,9) RC(0,10) RC(0,11) RC(0,12) RC(0,13) RC(0,14) RC(0,15) RC(0,16) RC(0,17)
+		RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4) RC(1,5) RC(1,6) RC(1,7) RC(1,8) RC(1,9) RC(1,10) RC(1,11) RC(1,12) RC(1,13) RC(1,14) RC(1,15) RC(1,16) RC(1,17)
+		RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4) RC(2,5) RC(2,6) RC(2,7) RC(2,8) RC(2,9) RC(2,10) RC(2,11) RC(2,12) RC(2,13) RC(2,14) RC(2,15) RC(2,16) RC(2,17)
+		RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,4) RC(3,5) RC(3,6) RC(3,7) RC(3,8) RC(3,9) RC(3,10) RC(3,11) RC(3,12) RC(3,13) RC(3,14) RC(3,15) RC(3,16) RC(3,17)
+		RC(4,0) RC(4,1) RC(4,2) RC(4,3) RC(4,4) RC(4,5) RC(4,6) RC(4,7) RC(4,8) RC(4,9) RC(4,10) RC(4,11) RC(4,12) RC(4,13) RC(4,14) RC(4,15) RC(4,16) RC(4,17)
+		RC(5,0) RC(5,1) RC(5,2) RC(5,3) RC(5,4) RC(5,5) RC(5,6) RC(5,7) RC(5,8) RC(5,9) RC(5,10) RC(5,11) RC(5,12) RC(5,13) RC(5,14) RC(5,15) RC(5,16) RC(5,17)
+		>;
+	};
```

Comment:
I believe this transform is unnecessary since it's a "perfect rectangle". Feel free to remove and test.

Reply:
This results in the following error:

`Error: nice_nano.dts.pre.tmp:532.1-19 Label or path default_transform not found`

Logically, we kinda need the columns and rows in order for the offset in <board>_right.overlay to work.

I tried removing the map, which it would be nice to get rid of, but that results in:

`devicetree error: 'map' is marked as required in 'properties:'`

Is there a way to use a minimal or placeholder map?

Reply:
The default transform you have here is inferred since it sequentially counts up in both rows and columns. No transform is necessary in this case. The error is due to the transform being undefined but still called on line 12 of this file.

Reply:
That kinda works (builds without error) if I also remove the reference from bfo9000_right.overlay

	&default_transform {
		col-offset = <9>;
	};

but then the whole layout falls apart. I get, for example, from the first five keys of the first column of either a board flashed as a left OR right half:

![image](https://user-images.githubusercontent.com/833516/101850935-b2d58380-3b28-11eb-9238-e939a168d645.png)

I'd love to get rid of all that boilerplate, but I'm not sure it's possible without some changes beyond the scope of this pull request.

Reply:
Yeah, I missed the point about the right half offset and how cols/rows are inferred without a transform. To remove the map it would require some minor code tweaks, which I agree is out of the scope of the PR. This works then!